R/QC3e.R

```diff
@@ -0,0 +1,553 @@
+#' @param verbose of tekstuele output uit script gewenst is (T) of niet (F). Staat
+#' standaard op F.
+#'
+#' @return metingen bestand met verdachte locaties/monsters. 
+#'
+#' @export
+#'
+
+QC3e <- function(d_metingen, 
+                 ph_naam = "pH", 
+                 hco3_naam = "HCO3", 
+                 ec_naam = "GELDHD",
+                 celcius = 25,
+                 add_bicarbonate = F, 
+                 add_phosphate = F, 
+                 verbose = F) {
+
+  # Check datasets op kolommen en unieke informatie
+  testKolommenMetingen(d_metingen)
+  d <- d_metingen
+  
+  # aanpassen van opgegeven namen hco3 & ph & EC naar hco3 & hv & ecv. 
+  # Dat zijn de drie namen die gebruikt worden in de berekengeleidbaarheid functie
+  d <- d %>%
+    mutate(parameter = str_replace(parameter, ec_naam, "ecv")) %>%
+    mutate(parameter = str_replace(parameter, hco3_naam, "hco3v")) %>%
+    mutate(parameter = str_replace(parameter, ph_naam, "hv"))
+
+
+  # gegevens apart zetten om later qcid weer toe te voegen
+  id <- d %>%
+    dplyr::filter(parameter == "ecv") %>%
+    dplyr::select(qcid, monsterid, jaar, maand, dag, putcode, filter) 
+  
+  # selecteer relevante ionen om mee te nemen in ionenbalans
+  # h is pH en hv is pH-veld
+  an <- c("cl", "hco3", "hco3v", "no3", "so4", "co3", "po4", "ptot")
+  cat <- c("al", "ca", "fe", "k", "mg", "mn", "nh4", "na", "zn", "h", "hv")
+  
+  #zelfde parameter namen gebruiken als in de functies voor het berekenen van de geleidbaarheid
+  d$parameter <- tolower(d$parameter) 
+  
+  # Dataset in juiste format zetten benodigd voor berekening ionenbalans en EC
+  # naar wide format en data goed zetten
+  res <- d %>%
+    # selecteer relevante ionen en pH ook meenemen -> omrekenen naar h30
+    dplyr::filter(parameter %in% c(an, cat, "ecv")) %>%
+    # alle NA's op 0 zetten, behalve pH en HCO3
+    dplyr::mutate(waarde_ib = ifelse(!parameter %in% c("h", "hv", "hco3", "hco3v") & is.na(waarde),
+                                     0, waarde)) %>%
+    # soms staat RG als NA, < of "", eerst NA veranderen in ""
+    dplyr::mutate(detectieteken = ifelse(is.na(detectieteken), "", 
+                                         detectieteken)) %>%
+    # waardes <RG niet meenemen maar op 0 zetten 
+    dplyr::mutate(waarde_ib = ifelse(!parameter %in% c("h", "hv") & detectieteken != "",
+                                     0, waarde_ib)) %>%
+    # als geen pH bekend is, dan is de pH 7 
+    dplyr::mutate(waarde_ib = ifelse(parameter %in% c("h", "hv") & is.na(waarde),
+                                     7, waarde_ib)) %>%
+    #  zet kolommen naar wide format
+    dplyr::select(-c(qcid, detectieteken, rapportagegrens, waarde)) %>%
+    tidyr::pivot_wider(., 
+                       names_from = parameter,
+                       names_glue = "x{parameter}",
+                       values_from = waarde_ib) 
+
+  # benodigde kolommen voor Patricks functies voor EC/ionenbalans
+  benodigde_col <- 
```

Comment:
xhco3 is geen benodigde kolom omdat die in de geleidbaarheids berekening op NA wordt gezet als hij niet aanwezig is. dit gebeurt ook voor xhco3v. het is wel noodzakelijk om in ieder geval één van de twee te hebben.